client/signup/steps/p2-site/index.jsx

```diff
@@ -165,12 +183,45 @@ class P2Site extends React.Component {
 
 							this.logValidationErrorToLogstash( error.error, errorMessage );
 						} else {
-							messages.site = {
-								[ error.error ]: error.message,
-							};
-
+							if ( SITE_TAKEN_ERROR_CODES.includes( error.error ) ) {
+								messages.site = {
+									[ ERROR_CODE_TAKEN_SITE ]: this.props.translate(
+										'Sorry, that site already exists! Here are some available alternatives:'
+									),
+								};
+							} else {
+								messages.site = {
+									[ error.error ]: error.message,
+								};
+							}
+
+							// We want to log the real error code and message. The above is formatted for the end user
+							// only.
 							this.logValidationErrorToLogstash( error.error, error.message );
 						}
+
+						if ( error.error && SITE_TAKEN_ERROR_CODES.includes( error.error ) ) {
+							WPCOM_SUBDOMAIN_SUFFIX_SUGGESTIONS.forEach( ( suffix ) => {
+								const suggestedSubdomain = `${ fields.site }${ suffix }.wordpress.com`;
+
+								wpcom
+									.domains()
+									.suggestions( {
+										quantity: 1,
+										query: suggestedSubdomain,
+										only_wordpressdotcom: true,
+									} )
+									.then( ( suggestionObjects ) => {
+										this.setState( {
+											suggestedSubdomains: [
+												...this.state.suggestedSubdomains,
+												get( suggestionObjects, '0.domain_name', null ),
+											],
+										} );
+									} )
+									.catch( () => {} );
```

Comment:
I understand that proper error handling might be out of scope here (and the only significant error that I can come up with is when someone searches for a blacklisted string like `wordpress` and gets a `invalid_query` error back). But maybe a quick improvement would be to split the message shown when the subdomain that user wants is taken:
```
Sorry, that site already exists! Here are some available alternatives:
```

The `Here are some available alternatives:` part could be shown when `this.state.suggestedSubdomains` is not empty. And if there's an error, you could update the state with it or something and show simply something like "Please try a different site name.".
I just want to avoid leaving the user with this confusing view (if it happens for some reason):

<img width="383" alt="Screenshot 2020-08-28 at 16 17 33" src="https://user-images.githubusercontent.com/3392497/91589892-ff1a0080-e949-11ea-942e-efda6804a914.png">